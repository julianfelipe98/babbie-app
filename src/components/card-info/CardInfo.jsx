import React from "react";
import "./CardInfo.css";
import { Button } from "@material-ui/core";
import logoPhone from "../../assets/cp2.png";

export const NavTools = ({ id, title, subtitle, paragraph }) => {
    const phone = (
        <div id="iconPhoneTool">
            <img id="imageCellphone" src={logoPhone} alt="logo-cellphone-2" />
        </div>
    );
    const info = (
        <div id="text-info-phone">
            <div id="text-tools-link">
                <p>{title}</p>
            </div>
            <h1>{subtitle}</h1>
            <h3>{paragraph}</h3>
        </div>
    );
    const buttonSearch = (
        <div id="text-info-phone">
            <div id="text-tools-link">
                <p>{title}</p>
            </div>
            <h1>{subtitle}</h1>
            <h3>{paragraph}</h3>
            <Button variant="outlined" color="primary" edge="end">
                Buscar historial medico
            </Button>
        </div>
    );

    return (
        <div id="navTools">
            {parseInt(id) === 1 && (
                <div id="info-tools-one">
                    {phone}
                    {info}{" "}
                </div>
            )}
            {parseInt(id) === 2 && (
                <div id="info-tools-one">
                    {info}
                    {phone}{" "}
                </div>
            )}
            {parseInt(id) === 3 && (
                <div id="info-tools-one">
                    {phone}
                    {buttonSearch}
                </div>
            )}
        </div>
    );
};
