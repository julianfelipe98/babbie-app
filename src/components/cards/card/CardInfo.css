#navTools {
  width: 100%;
  color: #000000;
  margin-top: 20px;
}
.card {
  display: flex;
  justify-content: space-evenly;
  text-align: start;
}
#text-info-phone {
  font-family: "Trebuchet MS", "Lucida Sans Unicode", "Lucida Grande",
    "Lucida Sans", Arial, sans-serif;
  align-self: center;
}
#imageCellphone {
  width: 20rem;
  height: auto;
  filter: drop-shadow(0 0 5px rgba(0,0,0,0.2));
}
#card-title {
  /* background-color: blue; */
  color: #497dfa;
  font-weight: 800;
}
h3 {
  color: #737373;
  font-weight: 400;
}
#subtitle-card{
  color: #252b42;
}
