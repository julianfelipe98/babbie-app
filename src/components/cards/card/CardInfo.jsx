import React from "react";
import "./CardInfo.css";
import { Button } from "@material-ui/core";
import logoPhone from "../../../assets/cp2.png";

export const CardInfo = ({ id, title, subtitle, paragraph, btn }) => {
  const phone = (
    <div id="iconPhoneTool">
      <img id="imageCellphone" src={logoPhone} alt="logo-cellphone-2" />
    </div>
  );
  const info = (
    <div id="text-info-phone">
      <div id="card-title">
        <p>{title}</p>
      </div>
      <h1 id="subtitle-card">{subtitle}</h1>
      <h3>{paragraph}</h3>
      {btn === "true" && (
        <Button variant="outlined" color="primary" edge="end">
          Buscar historial medico
        </Button>
      )}
    </div>
  );

  return (
    <>
      {(parseInt(id) === 0 || parseInt(id)===2)? (
        <div className="card">
          {phone}
          {info}
        </div>
      ) : (
        <div className="card">
          {info}
          {phone}
        </div>
      )}
    </>
  );
};
