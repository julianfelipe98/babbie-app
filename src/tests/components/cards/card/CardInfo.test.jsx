import { shallow } from "enzyme";
import { CardInfo } from "../../../../components/cards/card/CardInfo";

describe("CardInfo component tests", () => {
  const title = "title";
  const subtitle = "subtitle";
  const paragraph = "paragraph";
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <CardInfo id="1" title={title} subtitle={subtitle} paragraph={paragraph} />
    );
  });
  test("should show the correct title by props", () => {
    const textTitle = wrapper.find("p").at(0).text();
    console.log(wrapper.find("p"));
    expect(textTitle).toBe(title);
  });
  test("should show the correct subtitle by props", () => {
    const textSubtitle = wrapper.find("h1").at(0).text();
    expect(textSubtitle).toBe(subtitle);
  });
  test("should show the correct paragraph by props", () => {
    const textParagraph = wrapper.find("h3").at(0).text();
    expect(textParagraph).toBe(paragraph);
  });

  test("should render <CardInfo/> in the correct way", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
