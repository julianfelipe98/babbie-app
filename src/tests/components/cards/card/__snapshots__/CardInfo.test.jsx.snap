// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardInfo component tests should render <CardInfo/> in the correct way 1`] = `
<Fragment>
  <div
    className="card"
  >
    <div
      id="text-info-phone"
    >
      <div
        id="card-title"
      >
        <p>
          title
        </p>
      </div>
      <h1
        id="subtitle-card"
      >
        subtitle
      </h1>
      <h3>
        paragraph
      </h3>
    </div>
    <div
      id="iconPhoneTool"
    >
      <img
        alt="logo-cellphone-2"
        id="imageCellphone"
        src="cp2.png"
      />
    </div>
  </div>
</Fragment>
`;
